### DISCUSSION TITLE
TextMatch with Variable case-insensitive

### DISCUSSION THREAD

**Discussion by MentalGear** [testing-library/testing-library-docs#1166] [Q&A]
After going over the TextMatch page, I noticed there are regex matchers one can use like `/first name/i`, but I couldn't find out how to use these with a variable.

Example with fill function:
```ts
function fillInputElements(inputValuePair: { [id: string]: string }) {
    for (const [key, value] of Object.entries(inputValuePair)) {
        const inputEl = screen.getByRole('textbox', { name: key })
        inputEl.value = value
    }
}

fillInputElements({
    'First Name': 'John Simpleton2',
    'Last Name': 'Some LastName',
})
```

How to use the key with a case-insensitive match, or how can I make all role matches lowercase ?
I can't use exact: false as this would also introduce partial string matching which is something I don't want.

**Comment by eps1lon** [comment]
You can create a RegExp from a string. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp on how to do that.
